presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -122,16 +122,26 @@
 import javax.annotation.PreDestroy;
 import javax.inject.Inject;
 
-import java.util.List;
 import java.util.Set;
 
+import static java.util.Objects.requireNonNull;
+
 public class PlanOptimizers
+        implements PlanOptimizersProvider
 {
-    private final List<PlanOptimizer> optimizers;
```

Comment:
why not to collect them in constructor and simply run a for loop in `provide` method? Is it because that way it is easier to debug?